core/src/main/java/org/elasticsearch/index/mapper/TextFieldMapper.java

```diff
@@ -333,6 +333,7 @@ protected void parseCreateField(ParseContext context, List<IndexableField> field
             Field field = new Field(fieldType().name(), value, fieldType());
             fields.add(field);
         }
+        createFieldNamesField(context, fields);
```

Comment:
maybe use NormFieldExistsQuery since we are on a 7.1 snapshot now?

Reply:
I'll do this as a followup PR as it will require the Lucene 7.1 snapshot to be updated too

Reply:
+1